### PR TITLE
Implemented IRC joining/parting handlers

### PR DESCRIPTION
**Pull request by Zedjones**
Implemented handlers for joining and parting, only thing I'm unsure of is if [QUIT](https://tools.ietf.org/html/rfc2812#section-3.1.7) qualified as parting or if we treat that separately.

### CODE REVIEW

File: internal/handlers/irc/handlers.go [hunk 1]
```diff
@@ -6,6 +6,11 @@ import (
 	"github.com/lrstanley/girc"
 )
 
+const (
+	joinFmt = "<system> %s joins"
+	partFmt = "<system> %s parts"
```
Comment (jwflory) [inline internal/handlers/irc/handlers.go]: It's a stylistic preference, but I assume this is sent in Telegram when an IRC user parts/joins:

`<system> jwflory joins`

My preference would be something like:

`* jwflory joins`

It's my preference, but I won't make this suggestion blocking.

**Comment by Tjzabel** [comment]
As an update, @Zedjones has agreed to make the `QUIT` functionality before putting this PR through.

**Comment by Zedjones** [comment]
Also need to change format for message string to be more readable

File: internal/handlers/irc/handlers.go [hunk 2]
```diff
@@ -23,21 +29,46 @@ func connectHandler(c Client) func(*girc.Client, girc.Event) {
 	}
 }
 
-func privMsgHandler(c Client) func(*girc.Client, girc.Event) {
+/*
+messageHandler handles the PRIVMSG IRC event, which entails both private
+and channel messages. However, it only cares about channel messages
+*/
+func messageHandler(c Client) func(*girc.Client, girc.Event) {
```
Comment (Tjzabel) [inline internal/handlers/irc/handlers.go]: Neat, thanks for renaming and explaining

File: internal/handlers/irc/handlers.go [hunk 3]
```diff
@@ -23,21 +29,46 @@ func connectHandler(c Client) func(*girc.Client, girc.Event) {
 	}
 }
 
-func privMsgHandler(c Client) func(*girc.Client, girc.Event) {
+/*
+messageHandler handles the PRIVMSG IRC event, which entails both private
+and channel messages. However, it only cares about channel messages
```
Comment (Tjzabel) [inline internal/handlers/irc/handlers.go]: I'm guessing this means we only handle channel messages?

File: internal/handlers/irc/handlers.go [hunk 4]
```diff
@@ -23,21 +29,46 @@ func connectHandler(c Client) func(*girc.Client, girc.Event) {
...
+*/
+func messageHandler(c Client) func(*girc.Client, girc.Event) {
 	return func(gc *girc.Client, e girc.Event) {
-		formatted := fmt.Sprintf(c.MessageFormat, e.Source.Name, e.Params[1])
+		formatted := c.Settings.Prefix + e.Source.Name + c.Settings.Suffix + " " + e.Params[1]
```
Comment (Tjzabel) [inline internal/handlers/irc/handlers.go]: so this is fine, but we've actually historically made the `SUFFIX` variable set to `"> "` so we don't need to worry about the extra `+ " "` you have in there.

File: internal/handlers/irc/handlers.go [hunk 5]
```diff
@@ -23,21 +29,46 @@ func connectHandler(c Client) func(*girc.Client, girc.Event) {
...
 /*
 getHandlerMapping returns a mapping of girc event types to handlers
 */
 func getHandlerMapping() map[string]Handler {
 	return map[string]Handler{
 		girc.CONNECTED: connectHandler,
-		girc.PRIVMSG:   privMsgHandler,
+		girc.PRIVMSG:   messageHandler,
+		girc.PART:      partHandler,
+		girc.QUIT:      quitHandler,
```
Comment (Tjzabel) [inline internal/handlers/irc/handlers.go]: it would be *so great* if this was alphabetized

**Pull request merged**